src/common/translation.cpp

```diff
@@ -1887,12 +1887,28 @@ wxMsgCatalog *wxResourceTranslationsLoader::LoadCatalog(const wxString& domain,
     size_t mo_size = 0;
 
     const wxString resname = wxString::Format("%s_%s", domain, lang);
+    // Replace non-alphanumeric characters with `_`
+    wxString resname_safe = resname;
```

Comment:
This comment is not very useful because it describes what the code does (which is clear enough on its own) but not why does it do it, which is the important part. And, thinking about writing this comment, I've realized that we shouldn't need to ever sanitize the domain part, as it shouldn't contain anything non alphanumeric in the first place, AFAICS. Finally, the variable name is not great neither.

Combining all this together I'd do:
```suggestion
    // Language may contain non-alphabetic characters that are not allowed in the
    // resource names that must be valid identifiers, so sanitize the language
    // before using it as part of the resource name.
    wxString lang_sanitized = lang;
    ...
    const wxString resname = wxString::Format("%s_%s", domain, lang_sanitized);
```